### PR TITLE
🎉 Source Facebook Marketing: Add the option to fetch thumbnail image data.

### PR DESCRIPTION
**Pull request by zestyping**
## What
The Facebook ad_creatives stream contains a `thumbnail_url` field that points to a thumbnail image of the ad image or ad video.  However, the URL in that field is only valid for a short time; it expires after a few days.

This change allows you to optionally fetch the contents of the thumbnail image together with the ad_creatives stream.  The resulting data is encoded as a `data:` URL and placed in the new `thumbnail_data_url` field so you can easily use it as an image source URL.

## How
1. There's a new `fetch_thumbnail_images` setting in the Facebook Marketing source config.
2. When this setting is enabled, the `AdCreatives` stream post-processes the retrieved record by fetching the `thumbnail_url` (using the `requests` library) and encodes the data as a `data:` URL, which it stores in the `thumbnail_data_url` field.
3. A `thumbnail_data_url` field is added to the schema for the ad creatives stream, so that it can be returned in the Airbyte data.  However, this field is excluded when requesting data from Facebook, because it's not a field that Facebook knows about.

## 🚨 User Impact 🚨
There are no backward-incompatible changes; upgrading to this version of the Facebook Marketing source should have no effect on existing connections.  The `fetch_thumbnail_images` setting defaults to `False`, and the new behaviour is only enabled when the user changes the setting to `True`.

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [x] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</p>
</details>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
@misteryeo can you review the UX change this code makes?

File: airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py [hunk 1]
```diff
@@ -76,7 +76,9 @@ class Config:
         default_factory=pendulum.now,
     )
 
-    include_deleted: bool = Field(default=False, description="Include data from deleted campaigns, ads, and adsets.")
+    fetch_thumbnail_images: bool = Field(default=False, description="In each Ad Creative, fetch the thumbnail_url and store the result in thumbnail_data_url")
```
Comment (misteryeo) [inline airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py:79]: Is there any documentation that we can link to from Facebook that demonstrates how this works?

**Review by misteryeo — APPROVED** [review]
This LGTM - not a blocker but I'd be curious on my comment about documentation to see if we can hyperlink as part of the config description as I imagine not everyone is familiar with the concept outlined there.

**Review by marcosmarxm — APPROVED** [review]
thanks @zestyping

**Comment by zestyping** [comment]
Yay thank you!

**Pull request merged**